piptools/scripts/sync.py

```diff
@@ -58,6 +65,30 @@ def cli(dry_run, force, find_links, index_url, extra_index_url, no_index, quiet,
     to_install, to_uninstall = sync.diff(requirements, installed_dists)
 
     install_flags = []
+
+    # Install flags from requirements files.
+    for link in finder.find_links:
+        install_flags.extend(['-f', link])
+    if not finder.index_urls and isinstance(finder.index_urls, list):
+        install_flags.append('--no-index')
+    if finder.index_urls and isinstance(finder.index_urls, list):
+        install_flags.extend(['-i', finder.index_urls.pop(0)])
+    for extra_index in finder.index_urls:
+        install_flags.extend(['--extra-index-url', extra_index])
+    if finder.format_control.no_binary:
+        install_flags.extend(['--no-binary',
+                              ','.join(finder.format_control.no_binary)])
+    if finder.format_control.only_binary:
+        install_flags.extend(['--only-binary',
+                              ','.join(finder.format_control.only_binary)])
+    if finder.allow_all_prereleases:
+        install_flags.append('--pre')
+    if finder.process_dependency_links:
+        install_flags.append('--process-dependency-links')
+    for host_pattern in finder.secure_origins:
+        install_flags.extend(['--trusted-host', host_pattern[1]])
```

Comment:
AFAIK the `pip-compile` compiles only below flags to a `requirements.txt`:

- --index-url
- --extra-index-url
- --no-index
- --trusted-host
- --only-binary
- --no-binary

The rest of them would be ignored. Should the `pip-compile` also be fixed then?

Reply:
Perfect! I think it would be great if you implement the rest of flags here in this PR, to fix the whole pipeline `compile => sync`. However, i'm not insisting.